### PR TITLE
Add a peers_preferred option in Grin

### PR DESCRIPTION
**Pull request by quentinlesceller**
Add a peers_preferred option in Grin.toml.
Grin will attempt to connect to this peer no matter what.

### CODE REVIEW

**Comment by antiochp** [comment]
We already have the following - 

```
#hardcoded peer lists for allow/deny
#will *only* connect to peers in allow list
#peers_allow = ["192.168.0.1:13414", "192.168.0.2:13414"]
#will *never* connect to peers in deny list
#peers_deny = ["192.168.0.3:13414", "192.168.0.4:13414"]
```

Is this intended to be more flexible (and in addition to) the existing allow/deny lists?

**Comment by quentinlesceller** [comment]
Yes all peers are allowed. Plus we will peer to a selected list of peers. E.g. have a cluster of grin nodes that connect to each others and the rest of the network.

File: grin.toml [hunk 1]
```diff
@@ -33,6 +33,9 @@ db_root = ".grin"
 #
 #seeds = ["192.168.0.1:13414","192.168.0.2:13414"]
 
+#A list of preferred peers to connect to
```
Comment (antiochp) [inline grin.toml]: Can we nest this under p2p config so it sits alongside `peers_allow` and `peers_deny`?

**Comment by quentinlesceller** [comment]
Done @antiochp. I also picked up the TODO along the way :).

**Comment by antiochp** [comment]
:+1:

**Pull request merged**